From 5f5c906129a4f7b9fd85b4ea4eb0aadb182d4570 Mon Sep 17 00:00:00 2001
From: Julien Enselme <julien.enselme@ioda.net>
Date: Fri, 17 Feb 2017 17:00:09 +0100
Subject: [PATCH 8/8] Add GML2 serializer for Envelope

---
 src/ol/format/gml2.js | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/src/ol/format/gml2.js b/src/ol/format/gml2.js
index 1d7334b..dce8807 100644
--- a/src/ol/format/gml2.js
+++ b/src/ol/format/gml2.js
@@ -633,6 +633,18 @@ ol.format.GML2.prototype.writeSurfaceOrPolygonMember_ = function(node, polygon,
  * @private
  */
 ol.format.GML2.prototype.writeEnvelope = function(node, extent, objectStack) {
+  var context = objectStack[objectStack.length - 1];
+  var srsName = context['srsName'];
+  if (srsName) {
+    node.setAttribute('srsName', srsName);
+  }
+  var keys = ['lowerCorner', 'upperCorner'];
+  var values = [extent[0] + ' ' + extent[1], extent[2] + ' ' + extent[3]];
+  ol.xml.pushSerializeAndPop(/** @type {ol.XmlNodeStackItem} */
+      ({node: node}), ol.format.GML2.ENVELOPE_SERIALIZERS_,
+      ol.xml.OBJECT_PROPERTY_NODE_FACTORY,
+      values,
+      objectStack, keys, this);
 };
 
 
@@ -748,3 +760,15 @@ ol.format.GML2.SURFACEORPOLYGONMEMBER_SERIALIZERS_ = {
         ol.format.GML2.prototype.writeSurfaceOrPolygonMember_)
   }
 };
+
+
+/**
+ * @type {Object.<string, Object.<string, ol.XmlSerializer>>}
+ * @private
+ */
+ol.format.GML2.ENVELOPE_SERIALIZERS_ = {
+  'http://www.opengis.net/gml': {
+    'lowerCorner': ol.xml.makeChildAppender(ol.format.XSD.writeStringTextNode),
+    'upperCorner': ol.xml.makeChildAppender(ol.format.XSD.writeStringTextNode)
+  }
+};
-- 
2.9.3

